### PR TITLE
Updated CI config [SDK-1356]

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

- Added build and test jobs for **macOS** and **tvOS**
- Refactored existing config to be reusable with commands and executors

### References

- Needs https://github.com/auth0/SimpleKeychain/pull/82 to be merged first, or the tvOS job will fail

### Checklist

[X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
[X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
[ ] All existing and new tests complete without errors

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,40 +1,98 @@
...
-      - run:
-          name: Pod Lib Lint
-          command: bundle exec fastlane ios pod_lint
-      - run:
-          name: SwiftPM Test
-          command: swift test
+          grep -lR "shouldUseLaunchSchemeArgsEnv" *.* --null | xargs -0 sed -i '' -e 's/shouldUseLaunchSchemeArgsEnv = "YES"/shouldUseLaunchSchemeArgsEnv = "YES" codeCoverageEnabled = "YES"/g'
+      - restore_cache:
+          keys:
+            - carthage-cache-{{ checksum "Cartfile.resolved" }}
```
Comment (lbalmaceda) [inline .circleci/config.yml]: add a default key
```suggestion
            - carthage-cache-{{ checksum "Cartfile.resolved" }}
            - carthage-cache
```

Comment (Widcket) [reply .circleci/config.yml]: But if the file doesn't exist, there won't be a cache folder in the first place

Comment (Widcket) [reply .circleci/config.yml]: And the file is in version control, so it should be there

**Comment by lbalmaceda** [comment]
The file looks ok but tvOS tests are not passing. Let me know when it's fixed

**Pull request merged**